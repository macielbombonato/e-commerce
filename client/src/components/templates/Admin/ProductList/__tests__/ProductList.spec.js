import createTestContext from 'react-cosmos-test/enzyme';

import fixture from '../ProductList.fixture';

const { mount, getWrapper, get } = createTestContext({ fixture });

beforeEach(mount);

describe('#ProductList template', () => {
  it('renders component', () => {
    expect(getWrapper()).toMatchSnapshot();
  });

  it('redirects user to product creation page', () => {
    getWrapper('.btn-primary').simulate('click');

    expect(get('props').history.push).toBeCalledWith('/admin/product/save');
  });
});
