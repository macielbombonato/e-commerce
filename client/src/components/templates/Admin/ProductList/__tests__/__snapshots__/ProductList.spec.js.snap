// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ProductList template renders component 1`] = `
<ProductList
  history={
    Object {
      "push": [Function],
    }
  }
  product={
    Object {
      "list": Array [
        Object {
          "description": "description",
          "id": "123",
          "image": "",
          "name": "product",
          "price": 100,
        },
        Object {
          "description": "description",
          "id": "123",
          "image": "",
          "name": "product",
          "price": 100,
        },
        Object {
          "description": "description",
          "id": "123",
          "image": "",
          "name": "product",
          "price": 100,
        },
        Object {
          "description": "description",
          "id": "123",
          "image": "",
          "name": "product",
          "price": 100,
        },
      ],
    }
  }
>
  <Container
    tag="div"
  >
    <div
      className="container"
    >
      <h1
        className="display-4"
      >
        Product List
      </h1>
      <div
        className="clearfix"
      >
        <Button
          className="float-right"
          color="primary"
          onClick={[Function]}
          tag="button"
        >
          <button
            className="float-right btn btn-primary"
            onClick={[Function]}
            type="button"
          >
            Add New Product
          </button>
        </Button>
      </div>
      <br />
      <Table
        responsiveTag="div"
        tag="table"
      >
        <table
          className="table"
        >
          <thead>
            <tr>
              <th>
                Name
              </th>
              <th>
                Description
              </th>
              <th>
                Price
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              key="0"
            >
              <td>
                product
              </td>
              <td>
                description
              </td>
              <td>
                R$ 
                100.00
              </td>
            </tr>
            <tr
              key="1"
            >
              <td>
                product
              </td>
              <td>
                description
              </td>
              <td>
                R$ 
                100.00
              </td>
            </tr>
            <tr
              key="2"
            >
              <td>
                product
              </td>
              <td>
                description
              </td>
              <td>
                R$ 
                100.00
              </td>
            </tr>
            <tr
              key="3"
            >
              <td>
                product
              </td>
              <td>
                description
              </td>
              <td>
                R$ 
                100.00
              </td>
            </tr>
          </tbody>
        </table>
      </Table>
    </div>
  </Container>
</ProductList>
`;
